ui/src/views/Roles/ViewRoles/index.jsx

```diff
@@ -16,10 +17,23 @@ import HelpView from '../../../components/HelpView';
 }))
 export default class ViewRoles extends PureComponent {
   state = {
-    roleSearch: '',
+    roleSearch: this.props.history.location.search
+      ? parse(this.props.history.location.search.slice(1)).roleSearch
+      : '',
   };
```

Comment:
Turns out we don't need to use internal state anymore since all the state is stored in the URL:

```diff
diff --git a/ui/src/views/Roles/ViewRoles/index.jsx b/ui/src/views/Roles/ViewRoles/index.jsx
index d550edaae..e01599f03 100644
--- a/ui/src/views/Roles/ViewRoles/index.jsx
+++ b/ui/src/views/Roles/ViewRoles/index.jsx
@@ -16,25 +16,17 @@ import HelpView from '../../../components/HelpView';
   },
 }))
 export default class ViewRoles extends PureComponent {
-  state = {
-    roleSearch: this.props.history.location.search
-      ? parse(this.props.history.location.search.slice(1)).roleSearch
-      : '',
-  };
-
   handleRoleSearchSubmit = roleSearch => {
     const { location, history } = this.props;
     const query = parse(location.search.slice(1));
 
-    if (query.roleSearch !== roleSearch) {
-      query.roleSearch = roleSearch;
+    if (query.search !== roleSearch) {
+      query.search = roleSearch;
 
       history.push({
         search: stringify(query, { addQueryPrefix: true }),
       });
     }
-
-    this.setState({ roleSearch });
   };
 
   handleCreate = () => {
@@ -42,8 +34,9 @@ export default class ViewRoles extends PureComponent {
   };
 
   render() {
-    const { classes, description } = this.props;
-    const { roleSearch } = this.state;
+    const { classes, description, location } = this.props;
+    const query = parse(location.search.slice(1));
+    const roleSearch = query.search ? query.search : '';
 
     return (
       <Dashboard
```

Let's also do the same thing to the other view :)

Reply:
Great! Thanks for the advice. I just updated this part. Have a look!